internal/app/backend/backend_service.go

```diff
@@ -75,198 +70,194 @@ func (s *backendService) FetchMatches(req *pb.FetchMatchesRequest, stream pb.Bac
 		return err
 	}
 
-	// Send errors from the running go routines back to the FetchMatches go
-	// routine.  Must have size equal to number of senders so that if FetchMatches
-	// returns an error, additional errors don't block the go routine from
-	// finishing.
-	errors := make(chan error, 2)
 	mmfCtx, cancelMmfs := context.WithCancel(stream.Context())
-	startMmfs := func() error {
-		resultChan := make(chan mmfResult, 1)
+	// Closed when mmfs should start.
+	startMmfs := make(chan struct{})
+	proposals := make(chan *pb.Match)
+	wg := sync.WaitGroup{}
+	wg.Add(3)
 
-		err := doFetchMatchesReceiveMmfResult(mmfCtx, s.mmfClients, req, resultChan)
-		if err != nil {
-			// TODO: Log but continue case where mmfs were canceled once fully
-			// streaming.
-			return err
-		}
+	var synchronizeSendErr error
+	go func() {
+		defer wg.Done()
+		synchronizeSendErr = synchronizeSend(mmfCtx, proposals, syncStream)
+	}()
 
-		proposals, err := doFetchMatchesValidateProposals(mmfCtx, resultChan, 1)
-		if err != nil {
-			// TODO: Log but continue case where mmfs were canceled once fully
-			// streaming.
-			return err
+	var synchronizeRecvErr error
+	go func() {
+		defer wg.Done()
+		synchronizeRecvErr = synchronizeRecv(syncStream, stream, startMmfs, cancelMmfs)
+		cancelMmfs()
+	}()
+
+	var mmfErr error
+	go func() {
+		defer wg.Done()
+
+		select {
+		case <-mmfCtx.Done():
+			mmfErr = fmt.Errorf("Mmf was never started")
+			return
+		case <-startMmfs:
 		}
 
-	sendProposals:
-		for _, p := range proposals {
-			select {
-			case <-mmfCtx.Done():
-				logger.Warning("proposals from mmfs received too late to be sent to synchronizer")
+		mmfErr = callMmf(mmfCtx, s.cc, req, proposals)
+	}()
+
+	wg.Wait()
+
+	// TODO: Send mmf error in FetchSummary instead of erroring call.
+	if synchronizeSendErr != nil || synchronizeRecvErr != nil || mmfErr != nil {
+		logger.WithFields(logrus.Fields{
+			"synchronizeSendErr": synchronizeSendErr,
+			"synchronizeRecvErr": synchronizeRecvErr,
+			"mmfErr":             mmfErr,
+		}).Error("error(s) in FetchMatches call.")
+
+		return fmt.Errorf(
```

Comment:
Would it make more sense to use `status.Errorf` at here?

Reply:
The way I'm combining errors, not really.  It's not clear, eg, if there are different status codes what the combined one should be.  FetchSummary will alleviate this and this call shouldn't ever end with an error being returned.